optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/reproducer/TestGenCorruptedScoreReproducer.java

```diff
@@ -66,10 +66,10 @@ public boolean isReproducible(TestGenKieSessionJournal journal) {
         } catch (RuntimeException e) {
             if (e.getMessage() != null && e.getMessage().startsWith("No fact handle for ")) {
                 // this is common when removing insert of a fact that is later updated - not interesting
-                logger.debug("    Can't remove insert: {}", e.toString());
+                logger.debug("    Can't remove insert: {}", (Object) e);
```

Comment:
Unfortunately, this doesn't work as expected. The result looks like:
```
18:39:02.382 [main] DEBUG     Can't remove insert: {}
java.lang.IllegalStateException: No fact handle for cloudProcess_847
	at org.optaplanner.core.impl.score.director.drools.testgen.operation.TestGenKieSessionUpdate.invoke(TestGenKieSessionUpdate.java:80) ~[classes/:na]
	(full stack trace)
```
It seems that SLF4J is too [smart](https://www.slf4j.org/faq.html#paramException) and always detects whether the last parameter is an Exception even if you cast it to Object and even if it results in an unused placeholder.